openforcefield/typing/engines/smirnoff/io.py

```diff
@@ -263,7 +263,27 @@ def prepend_all_keys(d, char='@'):
                 for item in d:
                     prepend_all_keys(item)
 
+        # the "xmltodict" library defaults to print out all element attributes on separate lines
+        # unless they're prepended by "@"
         prepend_all_keys(smirnoff_data['SMIRNOFF'])
+
+        # Un-prepend "@" from "Author" and "Date" fields, as they SHOULD be printed on
+        # separate lines from the top-level SMIRNOFF tag
+        if '@Author' in smirnoff_data['SMIRNOFF'].keys():
+            smirnoff_data['SMIRNOFF']['Author'] = smirnoff_data['SMIRNOFF']['@Author']
+            del smirnoff_data['SMIRNOFF']['@Author']
```

Comment:
I think you can simplify the code by adding an `ignore_keys=['Author', 'Date']` argument in `prepend_all_keys(d, char='@', ignore_keys=frozenset())` and do something like
```python
if isinstance(d[key], list) or isinstance(d[key], dict):
    prepend_all_keys(d[key])
elif key not in ignore_keys:
    new_key = char + key
    d[new_key] = d[key]
    del d[key]
    prepend_all_keys(d[new_key])
```
This way we should be able to avoid also the re-ordering bit.

Reply:
Good idea. I've implemented that.